integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/CloudbreakTest.java

```diff
@@ -52,19 +63,42 @@ public CloudbreakTest() {
         testContext.putContextParam(PASSWORD, defaultUaaPassword);
     }
 
-    @BeforeTest
-    public void digestParameters(ITestContext testngContext) {
-        TestParameter.init();
+    public TestParameter getTestParameter() {
+        return testParameter;
+    }
 
-        YamlPropertiesFactoryBean yaml = new YamlPropertiesFactoryBean();
-        yaml.setResources(new FileSystemResource("application.yml"));
-        yaml.getObject().entrySet().stream().forEach(
-                entry -> TestParameter.put((String) entry.getKey(), entry.getValue().toString())
-        );
+    public void setTestParameter(TestParameter tp) {
+        testParameter = tp;
+    }
 
+    @BeforeTest(alwaysRun = true)
+    public void digestParameters(ITestContext testngContext) {
         LOGGER.info("CloudbreakTest load test parameters ::: ");
         if (testngContext != null) {
-            TestParameter.putAll(testngContext.getCurrentXmlTest().getAllParameters());
+            getTestParameter().putAll(testngContext.getCurrentXmlTest().getAllParameters());
+        }
+
+        LOGGER.info("Application.yml based parameters ::: ");
+        getTestParameter().putAll(getAllKnownProperties(environment));
+    }
+
+    private Map<String, String> getAllKnownProperties(Environment env) {
+        Map<String, String> rtn = new HashMap<>();
+        if (env instanceof ConfigurableEnvironment) {
+            for (PropertySource<?> propertySource : ((ConfigurableEnvironment) env).getPropertySources()) {
+                if (propertySource instanceof EnumerablePropertySource) {
+                    LOGGER.info("processing property source ::: " + propertySource.getName());
+                    for (String key : ((EnumerablePropertySource) propertySource).getPropertyNames()) {
+                        String value = propertySource.getProperty(key).toString();
+                        if (!"".equals(value)) {
```

Comment:
I would use StringUtils for similar checks, but this version is also null safe :)